tests/pfcwd/test_pfcwd_function.py

```diff
@@ -0,0 +1,518 @@
+                      'port_src': self.pfc_wd_test_port_id,
+                      'port_dst': dst_port,
+                      'ip_dst': self.pfc_wd_rx_neighbor_addr,
+                      'port_type': self.port_type,
+                      'wd_action': action}
+        log_format = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M:%S")
+        log_file = "/tmp/pfc_wd.PfcWdTest.{}.log".format(log_format)
+        ptf_runner(self.ptf, "ptftests", "pfc_wd.PfcWdTest", "ptftests", params=ptf_params,
+                   log_file=log_file)
+
+    def verify_other_queue(self):
+        """
+        Send traffic on the other queue and verify that the packets get forwarded
+        """
+        logger.info("Send packets via {} to verify other queue is not affected".format(self.pfc_wd_test_port))
+        if type(self.pfc_wd_test_port_ids) == list:
+            dst_port = "".join(str(self.pfc_wd_test_port_ids)).replace(',', '')
+        else:
+            dst_port = "[ " + str(self.pfc_wd_test_port_ids) + " ]"
+        ptf_params = {'router_mac': self.eth0_mac,
+                      'queue_index': self.pfc_queue_index - 1,
+                      'pkt_count': self.pfc_wd_test_pkt_count,
+                      'port_src': self.pfc_wd_rx_port_id[0],
+                      'port_dst': dst_port,
+                      'ip_dst': self.pfc_wd_test_neighbor_addr,
+                      'port_type': self.port_type,
+                      'wd_action': 'forward'}
+        log_format = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M:%S")
+        log_file = "/tmp/pfc_wd.PfcWdTest.{}.log".format(log_format)
+        ptf_runner(self.ptf, "ptftests", "pfc_wd.PfcWdTest", "ptftests", params=ptf_params,
+                   log_file=log_file)
+
+    def verify_other_pg(self):
+        """
+        Send traffic on the other PG and verify that the packets get forwarded
+        """
+        logger.info("Send packets to {} to verify other pg is not affected".format(self.pfc_wd_test_port))
+        if type(self.pfc_wd_rx_port_id) == list:
+            dst_port = "".join(str(self.pfc_wd_rx_port_id)).replace(',', '')
+        else:
+            dst_port = "[ " + str(self.pfc_wd_rx_port_id) + " ]"
+        ptf_params = {'router_mac': self.eth0_mac,
+                      'queue_index': self.pfc_queue_index - 1,
+                      'pkt_count': self.pfc_wd_test_pkt_count,
+                      'port_src': self.pfc_wd_test_port_id,
+                      'port_dst': dst_port,
+                      'ip_dst': self.pfc_wd_rx_neighbor_addr,
+                      'port_type': self.port_type,
+                      'wd_action': 'forward'}
+        log_format = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M:%S")
+        log_file = "/tmp/pfc_wd.PfcWdTest.{}.log".format(log_format)
+        ptf_runner(self.ptf, "ptftests", "pfc_wd.PfcWdTest", "ptftests", params=ptf_params,
+                   log_file=log_file)
+
+    def send_dummy_traffic(self):
```

Comment:
If the intent is to fill up the buffer, I would just rename this to `fill_buffer`. I think that makes the test flow more clear. :)